aiohttp/cookiejar.py

```diff
@@ -117,16 +121,38 @@ def __init__(
             )
         self._expire_heap: List[Tuple[float, Tuple[str, str, str]]] = []
         self._expirations: Dict[Tuple[str, str, str], float] = {}
+        self._cookies_loaded = threading.Event()
+        self._cookies_saved = threading.Event()
+        self._cookies_loaded.set()  # Initialized as set when no cookies need to be loaded
 
     def save(self, file_path: PathLike) -> None:
+        self._cookies_saved.clear()
+        loop = asyncio.get_event_loop()
+        loop.run_in_executor(None, self._save_cookies, file_path)
+        # Wait until saving is complete
+        self._cookies_saved.wait()
+
+    def _save_cookies(self, file_path: PathLike) -> None:
         file_path = pathlib.Path(file_path)
         with file_path.open(mode="wb") as f:
             pickle.dump(self._cookies, f, pickle.HIGHEST_PROTOCOL)
+        self._cookies_saved.set()
 
     def load(self, file_path: PathLike) -> None:
+        self._cookies_loaded.clear()
+        loop = asyncio.get_event_loop()
+        loop.run_in_executor(None, self._load_cookies, file_path)
+        # Wait until loading is complete
+        self._cookies_loaded.wait()
```

Comment:
This will block the event loop for the duration of the operation in the thread

Reply:
So if I understand this correctly then, the event loop is still blocked while the code is now being run in the executor so the net effect is no real change in behaviour except that HA is now unable to detect and warn of a blocking operation in the event loop?

Reply:
If there's not await or async for/with, then it's not yielding back to the event loop, so it'd wait here till the function completes.